core/collateral/engine.go

```diff
@@ -3526,22 +3526,13 @@ func (e *Engine) CreatePartyAMMsSubAccounts(
 	ctx context.Context,
 	party, subAccount, asset, market string,
 ) (general *types.Account, margin *types.Account, err error) {
-	generalID := e.accountID(noMarket, subAccount, asset, types.AccountTypeGeneral)
-	if _, ok := e.accs[generalID]; ok {
-		return nil, nil, errors.New("general sub account already exists")
-	}
-
-	marginID := e.accountID(market, subAccount, asset, types.AccountTypeMargin)
-	if _, ok := e.accs[marginID]; ok {
-		return nil, nil, errors.New("general sub account already exists")
-	}
 
-	_, err = e.CreatePartyGeneralAccount(ctx, subAccount, asset)
+	generalID, err := e.CreatePartyGeneralAccount(ctx, subAccount, asset)
```

Comment:
Is it not possible for a party to submit an AMM, cancel it, and submit a second one later on? In that case, it feels a bit off to call `CreatePartyGeneralAccount` and `CreatePartyMarginAccount`